test/xml_parser.ts

```diff
@@ -46,6 +56,20 @@ module.exports.parseAsync = function (assert: any) {
     assert.equal(++x, 1);
 };
 
+module.exports.parse_async_with_replace = function (assert: any) {
+    const filename = __dirname + "/../../test/fixtures/parser.xml";
+    const str = fs.readFileSync(filename, "utf8").replace(/[\r]+/g, '');
+
+    let x = 0;
+
+    libxml.parseXmlAsync(str, {replaceEntities: true, validateEntities: true}).then((doc) => {
+        // @ts-ignore
+        assert.equal(18, doc._ref.parseFlags);
```

Comment:
We shouldn't access `_ref` outside the core lib files.

Implement and call Document.getParseFlags() instead.